crates/pathfinder/src/rpc/types.rs

```diff
@@ -335,7 +338,7 @@ pub mod reply {
             }
 
             let resp = match other {
-                Request(ref s) => serde_json::from_str::<FailedResponse>(s),
+                Call(ref s) => serde_json::from_str::<FailedResponse>(&s.to_owned().to_string()),
```

Comment:
@koivunej it seems like this might be the last issue. i'm running into an ownership issue here and unclear how to solve it. i figured `.to_owned()` would do the trick but seems not. any suggestions?

```
temporary value is freed at the end of this statement
```

Reply:
So earlier this all was a convinience used from https://github.com/eqlabs/pathfinder/blob/83ba46d6b0554a7f33a72b07cefb11c6073f8567/crates/pathfinder/src/rpc.rs#L1514 ... Perhaps just comment this all out to get it compile, then try the `invalid_contract_address` test to see what kind of error is returned now, and perhaps adapt the idea of this `PartialEq` implementation to that?

Reply:
it is correct, CallError represents a JSON-RPC error object. 

`Error::Request` was essentially the same thing as `Error::Call` but call doesn't represent it as a String. So it was removed to make more clear that this variant is to represent a failed JSON-RPC call but not clear enough obviously :P

and you can match direct on the `code`, `message` and `optional data` if it's a `CallError::Custom`...

Reply:
Ok! I modified it to just return true for now. Do you wanna take it from here?

Reply:
I have to admit I've been continuing a bit ahead, with niklasad1's tip the PartialEq becomes easy to fix after all (see https://github.com/eqlabs/pathfinder/commit/d6b57827164ea46310f1a43aba6a85628ba73b44). I've continued until I found a possible bug we might need to fix before next release via rewriting more tests to use `assert_eq!(ErrorCode::$variant, error);` from the `get_err(s)` model. I've pushed my progress to https://github.com/eqlabs/pathfinder/tree/tarrance_updaterpsee but failed to create a PR.

Let's hope to get out of this long thread for the remaining issues. I'll continue next to test `rpc::tests::events::positional_args::get_events_with_invalid_page_size` more.